### ISSUE TITLE
Op coverage initial tracking in end-to-end run (totally 106 ops)

### ISSUE THREAD

**Issue by kiszk** [torch-spyre/torch-spyre#166]
# See #1080 and #940. 
This issue is obsolete.

### What feature would you like to be added?

Add torch-spyre eager runtime and torch.compile supports for enable PyTorch operations assuming the backend compiler has support for these ops.
The function of each op should be verified by a few limited test cases in [the file](https://github.com/torch-spyre/torch-spyre/blob/main/tests/_inductor/test_inductor_ops.py).

### Why is this feature needed?

This issue tracks PyTorch operation e2e enablements (From PyTorch code to Spyre binary) to run the following four models on Spyre device:

- [gpt-oss](https://huggingface.co/openai/gpt-oss-20b)
- [granite4-h](https://huggingface.co/ibm-granite/granite-4.0-h-small)
- [granite3-speech](https://huggingface.co/ibm-granite/granite-speech-3.3-8b)
- [granite3-vision](https://huggingface.co/ibm-granite/granite-vision-3.3-2b)

### Anything else we need to know?

Please create a sub-issue **before** you start  working for an operator. 
Please **close sub-issues and mark in the following list** once some tests in PyTorch pass.

List of torch operations (Please mark a check when you closed a PR) 
- [x] torch.add_
- [ ] torch.all
- [ ] torch.amp.autocast_mode._enter_autocast
- [ ] torch.amp.autocast_mode._exit_autocast
- [x] torch.arange
- [x] torch.bmm
- [ ] torch.cat
- [ ] torch.chunk
- [x] torch.clamp
- [x] torch.clone
- [x] torch.contiguous
- [ ] torch.conv1d
- [ ] torch.conv2d
- [x] torch.copy_
- [x] torch.cos
- [ ] torch.cumsum
- [ ] torch.div
- [x] torch.eq
- [ ] torch.exp
- [ ] torch.expand
- [ ] torch.expand_as
- [ ] torch.flatten
- [ ] torch.float
- [x] torch.full
- [ ] torch.functional.einsum
- [ ] torch.functional.split
- [x] torch.ge
- [ ] torch.getitem
- [ ] torch.gt
- [ ] torch.index_add
- [ ] torch.index_copy_
- [ ] torch.int
- [x] torch.invert ("~", operator.__invert__)
- [ ] torch.item
- [x] torch.le
- [ ] torch.long
- [ ] torch.masked_fill
- [ ] torch.masked_fill_
- [ ] torch.masked_scatter
- [ ] torch.matmul
- [ ] torch.max
- [x] torch.mean
- [x] torch.mul
- [x] torch.mul_
- [x] torch.ne
- [x] torch.neg
- [x] torch.new_ones
- [ ] torch.nn.attention._backend_from_string
- [x] torch.nn.attention._sdpa_kernel
- [ ] torch.nn.functional.batch_norm
- [x] torch.nn.functional.dropout
- [ ] torch.nn.functional.embedding
- [ ] torch.nn.functional.glu
- [ ] torch.nn.functional.layer_norm
- [ ] torch.nn.functional.multi_head_attention_forward
- [x] torch.nn.functional.silu
- [ ] torch.nn.functional.softmax
- [x] torch.nn.gelu
- [x] torch.nn.linear
- [ ] torch.nn.pad
- [ ] torch.nn.scaled_dot_product_attention
- [x] torch.nn.softplus
- [x] torch.numel
- [ ] torch.ones
- [ ] torch.ops.aten._assert_scalar.default
- [ ] torch.ops.aten.index
- [ ] torch.permute
- [ ] torch.pow
- [ ] torch.repeat
- [ ] torch.repeat_interleave
- [ ] torch.reshape
- [x] torch.rsqrt
- [ ] torch.scalar_tensor
- [ ] torch.scatter
- [ ] torch.scatter_
- [x] torch.sigmoid
- [x] torch.sin
- [ ] torch.softmax
- [ ] torch.sort
- [ ] torch.split
- [ ] torch.sub
- [ ] torch.sum
- [ ] torch.sym_sum
- [ ] torch.tensor
- [ ] torch.to
- [ ] torch.topk
- [x] torch.transpose
- [ ] torch.tril
- [ ] torch.triu
- [ ] torch.truediv
- [ ] torch.type_as
- [x] torch.unsqueeze
- [x] torch.view
- [ ] torch.where
- [ ] torch.zeros
- [ ] torch.zeros_like
- [ ] _operator.setitem
- [ ] torch._C._functorch._add_batch_dim
- [ ] torch._C._functorch._remove_batch_dim
- [ ] torch._C._functorch._vmap_decrement_nesting
- [ ] torch._C._functorch._vmap_increment_nesting
- [x] torch.\_\_and\_\_
- [ ] torch.\_\_eq\_\_
- [ ] torch._check_is_size
- [ ] torch._functorch.vmap.lazy_load_decompositions

The above list is generated from sets of PyTorch operations to be executed.

gpt-oss
```
torch.add
torch.amp.autocast_mode._enter_autocast
torch.amp.autocast_mode._exit_autocast
torch.bmm
torch.cat
torch.chunk
torch.clamp
torch.contiguous
torch.cos
torch.expand
torch.float
torch.getitem
torch.index_copy_
torch.matmul
torch.max
torch.mean
torch.mul
torch.nn.functional.dropout
torch.nn.functional.embedding
torch.nn.functional.softmax
torch.nn.linear
torch.pow
torch.repeat
torch.reshape
torch.rsqrt
torch.scatter_
torch.sigmoid
torch.sin
torch.sub
torch.sum
torch.to
torch.topk
torch.transpose
torch.unsqueeze
torch.view
torch.zeros
torch.zeros_like
```

granite4-h
```
_operator.setitem
torch.__eq__
torch._check_is_size
torch.add
torch.all
torch.arange
torch.cat
torch.chunk
torch.clamp
torch.clone
torch.contiguous
torch.conv1d
torch.copy_
torch.cumsum
torch.div
torch.eq
torch.exp
torch.expand
torch.flatten
torch.float
torch.full
torch.functional.split
torch.ge
torch.getitem
torch.gt
torch.index_add
torch.invert
torch.item
torch.long
torch.masked_fill
torch.mean
torch.mul
torch.mul_
torch.neg
torch.nn.functional.embedding
torch.nn.functional.silu
torch.nn.linear
torch.nn.pad
torch.nn.scaled_dot_product_attention
torch.nn.softplus
torch.ones
torch.ops.aten._assert_scalar.default
torch.permute
torch.pow
torch.repeat_interleave
torch.reshape
torch.rsqrt
torch.scatter
torch.softmax
torch.sort
torch.split
torch.sub
torch.sum
torch.sym_sum
torch.to
torch.topk
torch.transpose
torch.tril
torch.triu
torch.truediv
torch.type_as
torch.view
torch.zeros
torch.zeros_like
```

granite3-speech
```
_operator.setitem
torch._C._functorch._add_batch_dim
torch._C._functorch._remove_batch_dim
torch._C._functorch._vmap_decrement_nesting
torch._C._functorch._vmap_increment_nesting
torch.__and__
torch.__eq__
torch._functorch.vmap.lazy_load_decompositions
torch.add
torch.add_
torch.all
torch.amp.autocast_mode._enter_autocast
torch.amp.autocast_mode._exit_autocast
torch.arange
torch.cat
torch.chunk
torch.clone
torch.contiguous
torch.conv1d
torch.cos
torch.expand
torch.float
torch.functional.einsum
torch.getitem
torch.int
torch.le
torch.masked_fill_
torch.masked_scatter
torch.matmul
torch.mean
torch.mul
torch.ne
torch.neg
torch.new_ones
torch.nn.attention._backend_from_string
torch.nn.attention._sdpa_kernel
torch.nn.functional.batch_norm
torch.nn.functional.dropout
torch.nn.functional.embedding
torch.nn.functional.glu
torch.nn.functional.layer_norm
torch.nn.functional.silu
torch.nn.functional.softmax
torch.nn.gelu
torch.nn.linear
torch.nn.pad
torch.nn.scaled_dot_product_attention
torch.numel
torch.ones
torch.ops.aten.index
torch.permute
torch.pow
torch.reshape
torch.rsqrt
torch.sin
torch.sub
torch.sum
torch.sym_sum
torch.tensor
torch.to
torch.transpose
torch.truediv
torch.unsqueeze
torch.view
torch.where
```

granite3-vision
```
torch._C._functorch._add_batch_dim
torch._C._functorch._remove_batch_dim
torch._C._functorch._vmap_decrement_nesting
torch._C._functorch._vmap_increment_nesting
torch.__and__
torch.__eq__
torch._functorch.vmap.lazy_load_decompositions
torch.add
torch.add_
torch.amp.autocast_mode._enter_autocast
torch.amp.autocast_mode._exit_autocast
torch.arange
torch.cat
torch.contiguous
torch.conv2d
torch.cos
torch.expand
torch.expand_as
torch.flatten
torch.float
torch.getitem
torch.gt
torch.item
torch.le
torch.masked_scatter
torch.matmul
torch.mean
torch.mul
torch.neg
torch.new_ones
torch.nn.functional.embedding
torch.nn.functional.layer_norm
torch.nn.functional.multi_head_attention_forward
torch.nn.functional.silu
torch.nn.gelu
torch.nn.linear
torch.nn.scaled_dot_product_attention
torch.numel
torch.ops.aten.index
torch.pow
torch.repeat
torch.reshape
torch.rsqrt
torch.scalar_tensor
torch.sin
torch.sum
torch.sym_sum
torch.tensor
torch.to
torch.transpose
torch.truediv
torch.unsqueeze
torch.view
```

**Comment by kiszk** [comment]
These issues are the latest.
Q1: https://github.com/torch-spyre/torch-spyre/issues/1080
Q2: https://github.com/torch-spyre/torch-spyre/issues/940
